src/app/editors/collection-editor/collection-editor.component.ts

```diff
@@ -1,22 +1,173 @@
-import { Component, OnInit } from '@angular/core';
-import { Router } from '@angular/router';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
 import { HelperService } from '../../services/helper/helper.service';
-
+import * as _ from 'lodash-es';
+import { of } from 'rxjs';
 @Component({
   selector: 'app-collection-editor',
   templateUrl: './collection-editor.component.html',
   styleUrls: ['./collection-editor.component.scss']
 })
-export class CollectionEditorComponent implements OnInit {
-  public editorConfig = {};
-  constructor(private router: Router, public helperService: HelperService){}
+export class CollectionEditorComponent implements OnInit, OnDestroy {
+  public editorConfig: any;
+  public queryParams: any;
+  public content: any;
+  public channelData; any;
+  public hierarchyConfig;
+  constructor(private router: Router, private activatedRoute: ActivatedRoute, public helperService: HelperService){}
 
   ngOnInit(): void {
-    this.editorConfig = this.helperService.getConfig();
-    console.log(this.editorConfig, 'this.editorConfig');
+    this.queryParams = this.activatedRoute.snapshot.queryParams;
+    this.getContentDetails().subscribe((response) => {
+      this.content = _.get(response, 'result.content');
+      this.getChannel(_.get(response, 'result.content.channel'));
+      }, (error) => {
+         console.log('Fetching content details failed ::', error);
+      });
   }
+
   editorEventListener(event) {
     this.router.navigate(['editors/content-list/collection-editor']);
+  }
+
+  private getContentDetails() {
+    if (this.queryParams.identifier) {
+      const options: any = { params: { mode: 'edit' } };
+      return this.helperService.getContent(this.queryParams.identifier, options);
+    } else {
+      return of({});
+    }
+  }
+
+  getChannel(channelId) {
+    this.helperService.getChannel(channelId)
+      .subscribe((response) => {
+        this.channelData = _.get(response, 'result.channel');
+        this.getFrameWorkDetails();
```

Comment:
Why this is called here in sequence & why are we waiting till the channel response?